buildSrc/src/main/kotlin/io/spine/internal/gradle/github/pages/UpdateGitHubPages.kt

```diff
@@ -0,0 +1,403 @@
+ * the SSH private (encrypted `deploy_key_rsa`) and the public ("GitHub Pages publisher (Travis CI)"
+ * on GitHub) keys.
+ *
+ * Another requirement is an environment variable `REPO_SLUG`, which is set by the CI environment,
+ * such as `Publish` GitHub Actions workflow. It points to the repository for which the update
+ * is executed. E.g.:
+ *
+ * ```
+ *      REPO_SLUG: SpineEventEngine/base
+ * ```
+ *
+ * @see UpdateGitHubPagesExtension for the extension which is used to configure this plugin
+ */
+class UpdateGitHubPages : Plugin<Project> {
+
+    /**
+     * Root folder of the repository, to which this `Project` belongs.
+     */
+    private lateinit var rootFolder: File
+
+    /**
+     * Path to the temp folder used to gather the Javadoc output
+     * before submitting it to the GitHub Pages update.
+     */
+    private val javadocOutputPath = LazyTempPath("javadoc")
+
+    /**
+     * Path to the temp folder used checkout the original GitHub Pages branch.
+     */
+    private val checkoutTempFolder = LazyTempPath("repoTemp")
+
+    companion object {
+        /**
+         * The name of the task which updates the GitHub Pages.
+         */
+        const val taskName = "updateGitHubPages"
+
+        /**
+         * The name of the helper task to gather the generated Javadoc before updating GitHub Pages.
+         */
+        const val copyJavadoc = "copyJavadoc"
+
+        /**
+         * The name of the environment variable that contains the email to use for authoring
+         * the commits to the GitHub Pages branch.
+         */
+        const val formalGitHubPagesAuthorVar = "FORMAL_GIT_HUB_PAGES_AUTHOR"
+
+        /**
+         * The name of the environment variable containing the repository slug, for which
+         * the Gradle build is performed.
+         */
+        const val repoSlugVar = "REPO_SLUG"
+
+        /**
+         * The branch to use when pushing the updates to the documentation.
+         */
+        const val gitHubPagesBranch = "gh-pages"
+    }
+
+    /**
+     * Applies the plugin to the specified [project].
+     *
+     * If the project version says it is a snapshot, the plugin is not applied.
+     */
+    override fun apply(project: Project) {
+        val extension = UpdateGitHubPagesExtension.create(project)
+        project.extensions.add(UpdateGitHubPagesExtension::class, "updateGitHubPages", extension)
+        project.afterEvaluate {
+            val projectVersion = project.version.toString()
+            val isSnapshot = isSnapshot(projectVersion)
+            if (isSnapshot) {
+                println(
+                    "GitHub Pages update will be skipped since this project" +
+                            " is a snapshot: `${project.name}-${project.version}`."
+                )
+            } else {
+                registerTasks(extension, project)
+            }
+        }
+    }
+
+    private fun registerTasks(extension: UpdateGitHubPagesExtension, project: Project) {
```

Comment:
Please consider having this as an extension private function for `Project`.

Reply:
As discussed vocally, it's hardly doable due to some shared state.